analytical_engine/core/fragment/dynamic_fragment.h

```diff
@@ -344,15 +347,23 @@ class DynamicFragment
       vid_t new_ivnum = vm_ptr_->GetInnerVertexSize(fid_);
       vid_t new_ovnum = ovgid_.size();
       is_selfloops_.resize(new_ivnum);
-      ie_.reserve_vertices(new_ivnum);
-      oe_.reserve_vertices(new_ivnum);
-      this->ivnum_ = new_ivnum;
+      this->ivnum_ = new_ivnum;      
       if (ovnum_ != new_ovnum) {
         ovnum_ = new_ovnum;
         initOuterVerticesOfFragment();
       }
-      if (!edges_to_add.empty()) {
-        addEdges(edges_to_add);
+      ie_.add_vertices(new_ivnum - ivnum_, new_ovnum - ovnum_);
+      oe_.add_vertices(new_ivnum - ivnum_, new_ovnum - ovnum_);
+      if (this->directed_) {
+        ie_.add_reversed_edges(edges_to_add);
```

Comment:
We can not use `add_forward_edges` directly here since the edges_to_add could be contains edges that already existed and it just need to update the edges. `The add_forward/reverse_edges` would duplicate the edge

Reply:
You should check the edge exist or not and then update it or insert it, like `addEdges` do,

Reply:
btw, the `edges_to_add` could contains two same edges. In this scenairo, the correct strategy is insert with the first edge and update with second or other same edges.